controllers/resource/resource_generator.go

```diff
@@ -164,6 +179,59 @@ func (r *resourceGenerator) GetPodTemplate(doguResource *k8sv1.Dogu, dogu *core.
 	return podTemplate, nil
 }
 
+func getChownInitContainer(dogu *core.Dogu, doguResource *k8sv1.Dogu) (*corev1.Container, error) {
+	// Skip chown volumes with dogu-operator client because these are volumes from configmaps and read only.
+	filteredVolumes := filterVolumesWithClient(dogu.Volumes, doguOperatorClient)
+	if len(filteredVolumes) == 0 {
+		return nil, nil
+	}
+
+	initCommand := ""
+	for index, volume := range filteredVolumes {
+		uid, err := strconv.ParseInt(volume.Owner, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse owner id %s from volume %s: %w", volume.Owner, volume.Name, err)
+		}
+		gid, err := strconv.ParseInt(volume.Group, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse group id %s from volume %s: %w", volume.Group, volume.Name, err)
+		}
+
+		if uid > 0 && gid > 0 {
+			mkdirCommand := fmt.Sprintf("mkdir -p %s", volume.Path)
```

Comment:
Since we execute a shell here and all arguments reside in a single string I think it is necessary to handle directories containing whitespace (I know this sounds weird but see it as a precautionary measure)

A `filepath.Split(volume.Path)` or similar may be in order to handle/escape directories with such a format.

What would the tests look like if we had a directory like `/opt/dogu/hello world/file.txt`?